File: algorithms/cpp/102.cpp
#include <iostream>
#include <vector>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution
{
private:
    void go(vector< vector<int> > &result, struct TreeNode *root, int depth)
    {
        if ( !root )
            return;
        if ( depth >= (int)result.size() )
        {
            vector<int> level;
            level.push_back(root->val);
            result.push_back(level);
        }
        else
        {
            result[depth].push_back(root->val);
        }
        go(result, root->left, depth+1);
        go(result, root->right, depth+1);
    }
public:
    vector< vector<int> > levelOrder(TreeNode* root)
    {
        vector< vector<int> > result;
        go(result, root, 0);
        return result;
    }
};

int main()
{
    return 0;
}

File: algorithms/python/102.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        rls = []
        if not root:
            return rls
        self.helper(root, 0, rls)
        return rls

    def helper(self, root, depth, rls):
        if depth >= len(rls):
            rls.append([])
        rls[depth].append(root.val)
        if root.left:
            self.helper(root.left, depth+1, rls)
        if root.right:
            self.helper(root.right, depth+1, rls)
